### PR TITLE
Migrate contributing.md to docs site

### LINKED ISSUE #1180: Move and expand contributing.md content into docs site
## Documentation Change

There is a lot going in our contributing.md file and we're thinking of adding more. Moreover, we have several different audiences for this file (docs writers, testers, and developers), and we might be able to reach more of them through the jrnl.sh site.

It seems like it would be a good move to create a new section in the documentation for contributors, with subsections for each of these audiences.

As part of this, the developer documentation could be expanded to include some information on the project's architecture and our process for pinning and resolving GitHub issues.

After this work is done, the contributing.md shouldn't be removed, but it should mainly just link to this new documentation.

### PR DESCRIPTION
**Pull request by micahellison**
Fixes #1180. When this is done, it should be a little more user-friendly for non-developers to contribute.

Probably want to split up the contributing.md file into multiple files.

This adds some elements to the docs site that look a little funny - top-level non-link headings to divide up the docs, and lower-level headings when the user clicks on a link in a table of contents to get to an anchor. Need to style those appropriately before merging this PR.

### Checklist

- [x] I have read the [contributing doc](https://github.com/jrnl-org/jrnl/blob/develop/CONTRIBUTING.md).
- [x] I have included a link to the relevant issue number.
- [x] I have tested this code locally.
- [x] I have checked to ensure there aren't other open [pull requests](../pulls)
  for the same issue.
- ~~I have written new tests for these changes, as needed.~~
- [x] All tests pass.

### CODE REVIEW

**Comment by micahellison** [comment]
This is failing due to the issue that #1298 fixed. The next commit to this PR should make this green again.

**Comment by micahellison** [comment]
I think I've got the content in a good place, but there are some style changes that need to happen:

- [x] Nested lists need to indent (search for "poetry shell" for an example)
- [x] Navigation sub-menu contrast is bad -- white on gray. Click one of the links in the table contents then look to the left to see it.
- [ ] Navigation categories ("USER GUIDE" and "CONTRIBUTING") have really similar formatting to the navigation links, though the categories aren't clickable. Might be nice to make them a little more different.

**Comment by micahellison** [comment]
@wren could you style the h3s in the navs? See [here](https://jrnl--1266.org.readthedocs.build/en/1266/formats/#data-formats) for an example.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,125 +2,4 @@
...
-* [help wanted](https://github.com/jrnl-org/jrnl/labels/help%20wanted)
-* [bug](https://github.com/jrnl-org/jrnl/labels/bug)
-* [enhancement](https://github.com/jrnl-org/jrnl/labels/enhancement)
-
-### A note for new programmers and programmers new to python
-
-Although jrnl has grown quite a bit since its inception, the overall complexity (for an end-user program) is fairly low, and we hope you'll find the code easy enough to understand.
-
-If you have a question, please don't hesitate to ask! Python is known for its welcoming community and openness to novice programmers, so feel free to fork the code and play around with it! If you create something you want to share with us, please create a pull request. We never expect pull requests to be perfect, idiomatic, instantly mergeable code. We can work through it together!
+See the "Contributing" section of the [jrnl documentation site](https://jrnl.sh).
```
Comment (wren) [inline CONTRIBUTING.md]: Can this be a link to `docs/contributing.md`?

**Review by wren — APPROVED** [review]
🧧

**Pull request merged**